packages/core/src/machines/index.ts

```diff
@@ -635,10 +635,10 @@ export const createAuthMachine = ({
           }),
 
         autoSignIn: async () => {
-          if (typeof window !== 'undefined') {
-            const location = window.location
-            if (location.hash) {
-              const params = new URLSearchParams(location.hash.slice(1))
+          if (typeof window !== 'undefined' && window.location) {
+            const hash = window.location.hash
+            if (hash) {
+              const params = new URLSearchParams(hash.slice(1))
               const refreshToken = params.get('refreshToken')
               if (refreshToken) {
```

Comment:
We can make an early exit here similar to how we would return if `hash` is `undefined`.

The only reason I'd like to convert this is because we have too many nested if statements here.

Reply:
ok